name.abuchen.portfolio/src/name/abuchen/portfolio/model/AccountTransaction.java

```diff
@@ -45,6 +46,53 @@ public String toString()
         }
     }
 
+    public static class AccountTransactionBuilder
+    {
+        private Type txType;
+        private LocalDateTime dateTime;
+        private String currencyCode;
+        private long amount;
+        private Security security;
+
+        public AccountTransactionBuilder(Type txType)
+        {
+            this.txType = txType;
+        }
+
+        public AccountTransactionBuilder transactionAt(LocalDateTime dateTime)
+        {
+            this.dateTime = dateTime;
+            return this;
+        }
+
+        public AccountTransactionBuilder withCurrency(String currencyCode)
+        {
+            this.currencyCode = currencyCode;
+            return this;
+        }
+
+        public AccountTransactionBuilder withAmountOf(long amount)
+        {
+            this.amount = amount;
+            return this;
+        }
+
+        public AccountTransactionBuilder forSecurity(Security security)
+        {
+            this.security = security;
+            return this;
+        }
+
+        public AccountTransaction build()
+        {
+            Objects.requireNonNull(dateTime, "dateTime is a required field"); //$NON-NLS-1$
+            Objects.requireNonNull(currencyCode, "currencyCode is a required field"); //$NON-NLS-1$
+            Objects.requireNonNull(amount, "amount is a required field"); //$NON-NLS-1$
+            Objects.requireNonNull(security, "security is a required field"); //$NON-NLS-1$
```

Comment:
Missing translation and capitalization

Reply:
Why? This exception should be part of the log file, not for the user.

Reply:
I would recommend leaving out the strings completely, as they are optional. Perhaps even leave out the tests altogether …